pkg/snapshot/copier/copier.go

```diff
@@ -0,0 +1,135 @@
+	if err != nil {
+		return nil, nil, fmt.Errorf("could not get source snapstore: %v", err)
+	}
+
+	destSnapStore, err := snapstore.GetSnapstore(destSnapStoreConfig)
+	if err != nil {
+		return nil, nil, fmt.Errorf("could not get destination snapstore: %v", err)
+	}
+
+	return sourceSnapStore, destSnapStore, nil
+}
+
+// NewCopierConfig creates a new copier config.
+func NewCopierConfig() *brtypes.CopierConfig {
+	sourceSnapstoreConfig := snapstore.NewSnapstoreConfig()
+	sourceSnapstoreConfig.IsSource = true
+	return &brtypes.CopierConfig{
+		SourceSnapstore: sourceSnapstoreConfig,
+	}
+}
+
+// Copier can be used to copy backups from a source to a destination store.
+type Copier struct {
+	logger          *logrus.Entry
+	sourceSnapStore brtypes.SnapStore
+	snapStore       brtypes.SnapStore
+}
+
+// NewCopier creates a new copier.
+func NewCopier(sourceSnapStore brtypes.SnapStore, snapStore brtypes.SnapStore, logger *logrus.Entry) *Copier {
+	return &Copier{
+		logger:          logger.WithField("actor", "copier"),
+		sourceSnapStore: sourceSnapStore,
+		snapStore:       snapStore,
+	}
+}
+
+// Run executes the copy command.
+func (c *Copier) Run() error {
+	return c.CopyBackups()
+}
+
+// CopyBackups copies all backups from the source store to the destination store.
+func (c *Copier) CopyBackups() error {
+	// Get source backups
+	c.logger.Info("Getting source backups...")
+	sourceBackups, err := miscellaneous.GetAllBackups(c.sourceSnapStore)
+	if err != nil {
+		return fmt.Errorf("could not get source backups: %v", err)
+	}
+
+	// If there are no source backups, do nothing
+	if len(sourceBackups) == 0 {
+		c.logger.Info("No source backups found")
+		return nil
+	}
+
+	// Get destination snapshots and build a map keyed by name
+	c.logger.Info("Getting destination snapshots...")
+	snapshots, err := c.snapStore.List()
+	if err != nil {
+		return fmt.Errorf("could not get destination snapshots: %v", err)
+	}
+	snapshotsMap := make(map[string]*brtypes.Snapshot)
+	for _, snapshot := range snapshots {
+		snapshotsMap[snapshot.SnapName] = snapshot
+	}
+
+	for _, backup := range sourceBackups {
+		// Copy the full snapshot (if not already copied)
+		if _, ok := snapshotsMap[backup.FullSnapshot.SnapName]; !ok {
+			c.logger.Infof("Copying full snapshot %s...", backup.FullSnapshot.SnapName)
+			if err := c.copySnapshot(backup.FullSnapshot); err != nil {
+				return err
+			}
+		} else {
+			c.logger.Infof("Skipping full snapshot %s as it already exists", backup.FullSnapshot.SnapName)
+		}
+
+		// Copy all delta snapshots (if not already copied)
+		for _, deltaSnapshot := range backup.DeltaSnapshotList {
+			if _, ok := snapshotsMap[deltaSnapshot.SnapName]; !ok {
+				c.logger.Infof("Copying delta snapshot %s...", deltaSnapshot.SnapName)
+				if err := c.copySnapshot(deltaSnapshot); err != nil {
+					return err
+				}
+			} else {
+				c.logger.Infof("Skipping delta snapshot %s as it already exists", deltaSnapshot.SnapName)
+			}
+		}
```

Comment:
Please see the comment [above](https://github.com/gardener/etcd-backup-restore/pull/369/files#diff-d67a01c5aa3ff7bb12c7e9945be604ac31bd8a766f5962760de4d8223e639aedR66-R67).